src/util/piped_process.h

```diff
@@ -0,0 +1,70 @@
+/// \file Subprocess communication with pipes
+///
+/// \author Diffblue Ltd.
+
+#ifndef CPROVER_UTIL_PIPED_PROCESS_H
+#define CPROVER_UTIL_PIPED_PROCESS_H
+
+class piped_processt
+{
+public:
+  /// Enumeration to keep track of child process state.
+  enum class process_statet
+  {
+    NOT_CREATED,
+    CREATED,
+    STOPPED,
+    ERROR
+  };
+
+  enum class process_send_responset
+  {
+    SUCCEEDED,
+    FAILED,
+    ERROR
+  };
+
+  /// Send a string message (command) to the child process.
+  /// \param message The string message to be sent.
+  process_send_responset send(const std::string &message);
+  /// Read a string from the child process' output.
+  std::string receive();
+  /// Wait until a string is available and read a string from the child
+  /// process' output.
+  std::string wait_receive();
+
+  /// Get child process status.
+  process_statet get_status();
+
+  /// See if this process can receive data from the other process.
+  /// \param timeout Amount of time to wait before timing out and
```

Comment:
:question: What unit of time is `timeout` in? This should be documented, if not part of the parameter name.